### DISCUSSION TITLE
How do I sort a list by a word that appears multiple times in it?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#8053] [General] [answered]
The new todo.txt https://github.com/helix-editor/helix/pull/7835 format auto sorts on save alphabetically, but I also want the option to sort by via the `+` items too:
```
(A) tidy +home
call bob +phone
call alice +phone
something else +whatever
wash up +home
```
How can I sort so all the `+home` items are at the top rather than (A)? Can't find this in the docs anywhere.

**Comment by pascalkuthe** [comment]
the sorting on save is done using the UNIX sort program so you need to look at the manual for UNIX sort. alternatively you can simply write your own little program (small rust/go binary or a python script for example) to sort stdin however you want (and output back to stdout)

**Reply by David-Else** [reply]
Cheers, but I was looking at a way to sort it in Helix. I am not worried about the sort on save, but I would like to look at my list in different orders.

I can select everything, use `alt-s` to split per line, but then I need to select `+home` and sort the list using Helix so that all the lines with `+home` are at the top.

**Reply by pascalkuthe** [reply]
yeah that's not possible with the builtin sort command but you can still use `:pipe` to do the same thing I described

**Reply by David-Else** [reply, marked as answer]
Cheers! The answer is to actually sort via `@` and `+` instead, best results :)

```
%:pipe sort -t@ -k2
%:pipe sort -t+ -k2
```
The -t option is used to specify the field separator character. It allows you to define a custom delimiter character to separate fields in the input file. By default, sort uses whitespace as the field separator. With the -t option, you can specify a different character to be used as the field separator.

In Unix sort, the "-k2" option is used to specify the key field for sorting. It indicates that the sorting should be done based on the second field (column) of the input.

**Reply by mweihs** [reply]
A quick nitpick: in Unix sort the option "-k" takes a start and stop position. If no stop postion is given it defaults to "the end of the line". So to sort only on the second field, write "-k2,2".

**Reply by David-Else** [reply]
Thanks @mweihs! Nice update.

**Reply by David-Else** [reply]
I wonder if it is possible to sort again after the plain `sort` piping out the result, keeping the original order when the line starts with the same priority, for example `(A)`, but then sorting on `@`,  so rather than:

```
(A) task 1 @a  
(A) task 2 @c 
(A) task 3 @b < note a, c, b
(B) task 4 @a 
```
we ended up with:
```
(A) task 1 @a  
(A) task 3 @b 
(A) task 2 @c < note a, b, c
(B) task 4 @a < note original (A) > (B) order is retained
```
I have played around with the `sort` `-s` stable sort and a bit of `awk` and `sed` but can't get anything working. I don't want to make a bash script specially, just a simple one liner. Maybe you know of a simple way of doing it? I don't know these tools at all.

It would be a nice update the current default as all the `+` OR `@` contexts would line up making it easier to take action on a group of task with a certain priority.
